feat: fully implement Player#sendFakeBlock

Timongcraft opened the pull request:
## Overview
This PR improves the `Player#sendFakeBlock` implementation in Bukkit to support non structure-blocks (used by the CUI).

## Description
Replaces the structure block specific implementation with a generic implementation to allow API users to send fake blocks containing NBT such as player heads, banners etc.
Also brings the sendFakeOP method up the Player to separate from the 
sendFakeBlock method.
### Submitter Checklist
- [x] Make sure you are opening from a topic branch (**/feature/fix/docs/ branch** (right side)) and not your main branch.
- [x] Ensure that the pull request title represents the desired changelog entry.
- [x] New public fields and methods are annotated with `@since TODO`.
- [x] I read and followed the [contribution guidelines](https://github.com/IntellectualSites/.github/blob/main/CONTRIBUTING.md).

Timongcraft committed:
feat: fully implement Player#sendFakeBlock

dordsor21 commented on worldedit-bukkit/src/main/java/com/sk89q/worldedit/bukkit/BukkitPlayer.java:
```diff
@@ -52,10 +52,16 @@
 import com.sk89q.worldedit.world.gamemode.GameMode;
 import com.sk89q.worldedit.world.gamemode.GameModes;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
```
I would avoid this; either we should use a record or just Map.Entry, no need to add 3rd party dependencies wher it's not really needed (_especially_ apache commons as that can _massively_ increase the jar size)

Timongcraft committed:
chore: replace apache's pair usage with map entry

Timongcraft committed:
chore: update 1.21.11

Timongcraft committed:
chore: use extent instead of accessing the world directly
-> fix: rare exception when using CraftBlockEntityState#getSnapshotNBT async (with latest 1.21.10 Paper builds?)

Timongcraft committed:
fix: cui not showing for non-opped players
 -> decupled from fake block handling
chore: up deprecation version

Timongcraft committed:
Merge remote-tracking branch 'upstream/main' into feat/fully-impl-fake-block

Timongcraft committed:
chore: update 26.1

Timongcraft committed:
chore: update 26.2

Timongcraft committed:
fix: remove old 1.20 adapters leftovers (I prob. broke the merge somehow)

Timongcraft committed:
chore(jd): fixup and improve

Timongcraft commented on worldedit-core/src/main/java/com/sk89q/worldedit/entity/Player.java:
```diff
@@ -342,6 +342,8 @@ default boolean trySetPosition(Vector3 pos, float pitch, float yaw) {
     <B extends BlockStateHolder<B>> void sendFakeBlock(BlockVector3 pos, @Nullable B block);
 
+    void sendFakeOP();
+
```
> [...]  breaking change for any external Player implementations

not a concern
